Project fails to link

Looks like a few things slipped through the net with the latest commits:

Warning:
```
Button.cpp: In member function ‘bool Button::pressed(const sf::Event&)’:
Button.cpp:24:1: warning: control reaches end of non-void function [-Wreturn-type]
```

Link errors:
```
Snake.cpp.o: In function `Snake::respawn()':
Snake.cpp:(.text+0x117b): undefined reference to `Score::update(int)'
Snake.cpp.o: In function `Snake::addToSize(int)':
Snake.cpp:(.text+0x11d4): undefined reference to `Score::update(int)'
Main.cpp.o: In function `main':
Main.cpp:(.text+0x2e9): undefined reference to `Score::Score(std::string)'
Main.cpp:(.text+0x5f5): undefined reference to `Menu::Menu(sf::RenderWindow&, std::initializer_list<Button>)'
Main.cpp:(.text+0x6b6): undefined reference to `Menu::handleEvent(sf::Event const&)'
Main.cpp.o: In function `Score::~Score()':
Main.cpp:(.text._ZN5ScoreD2Ev[_ZN5ScoreD5Ev]+0x13): undefined reference to `vtable for Score'
Main.cpp.o: In function `Menu::~Menu()':
Main.cpp:(.text._ZN4MenuD2Ev[_ZN4MenuD5Ev]+0x13): undefined reference to `vtable for Menu'
```

--- a/SnakeFML/CMakeLists.txt
+++ b/SnakeFML/CMakeLists.txt
@@ -15,6 +15,8 @@ set(HDRS
     SnakeFML/MouseSpawner.hpp
     SnakeFML/Button.hpp
     SnakeFML/World.hpp
+    SnakeFML/Score.hpp
+    SnakeFML/Menu.hpp
 )
 
 set(SRCS
@@ -24,6 +26,8 @@ set(SRCS
     SnakeFML/Main.cpp
     SnakeFML/Button.cpp
     SnakeFML/World.cpp
+    SnakeFML/Score.cpp
+    SnakeFML/Menu.cpp
 )
 
 add_executable("${PROJECT_NAME}-exec" ${HDRS} ${SRCS})

--- a/SnakeFML/SnakeFML/Button.cpp
+++ b/SnakeFML/SnakeFML/Button.cpp
@@ -10,10 +10,10 @@ Button::Button(std::function<void()> clickCallback, std::string buttonGraphic) :
 
 bool Button::pressed(const sf::Event& event)
 {
-	switch (event.type) 
+	switch (event.type)
 	{
 	case sf::Event::MouseButtonPressed:
-		if (getGlobalBounds().contains(event.mouseButton.x, event.mouseButton.y)) 
+		if (getGlobalBounds().contains(event.mouseButton.x, event.mouseButton.y))
 		{
 			m_clickCallback();
 			return true;
@@ -21,5 +21,6 @@ bool Button::pressed(const sf::Event& event)
 	default:
 		break;
 	}
+	return false;
 }
 

--- a/SnakeFML/SnakeFML/Main.cpp
+++ b/SnakeFML/SnakeFML/Main.cpp
@@ -11,7 +11,7 @@
 int main()
 {
 	//state
-	enum gameState 
+	enum gameState
 	{
 		MENU,
 		INGAME,
@@ -25,7 +25,7 @@ int main()
 	icon.loadFromFile("Icon.png");
 	window.setIcon(icon.getSize().x, icon.getSize().y, icon.getPixelsPtr());
 	sf::Clock	dtClock;
-	
+
 	//Objects
 	Score			score("KBZipaDeeDooDah.ttf");
 	World			world("Grass.png", { 1000,1000 });
@@ -35,7 +35,7 @@ int main()
 	Button			quitButton([&]() {currentState = EXITING; }, "QuitButton.png");
 	Menu			menu(window,{ playButton, quitButton });
 
-	
+
 
 	while (window.isOpen())
 	{
@@ -53,7 +53,7 @@ int main()
 				currentState = EXITING;
 			}
 
-			switch (currentState) 
+			switch (currentState)
 			{
 			case MENU:
 				menu.handleEvent(event);
@@ -71,6 +71,7 @@ int main()
 						break;
 					}
 				}
+				break;
 			default:
 				break;
 			}
@@ -91,7 +92,7 @@ int main()
 		window.draw(world);
 		spawner.draw();
 		window.draw(snake);
-	
+
 		//default view for score (and menu, if needed)
 		window.setView(window.getDefaultView());
 		window.draw(score);

